src/mvc/collection.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Collections are ordered sets of models.
+ *
+ * Any event that is triggered on a model in a collection will also be triggered on the collection directly, for
+ * convenience.
+ *
+ * @class Collection
+ */
+
+CKEDITOR.define( [ 'emitter', 'utils' ], function( EmitterMixin, utils ) {
+	function Collection() {
+		/**
+		 * The internal list of models in the collection.
+		 *
+		 * @property _models
+		 * @private
+		 */
+		Object.defineProperty( this, '_models', {
+			value: []
+		} );
+
+		/**
+		 * The number of items available in the collection.
+		 *
+		 * @property length
+		 */
+		Object.defineProperty( this, 'length', {
+			get: function() {
+				return this._models.length;
+			}
+		} );
+	}
+
+	/**
+	 * @inheritdoc utils#extend
+	 */
+	Collection.extend = utils.extendMixin;
+
+	utils.extend( Collection.prototype, EmitterMixin, {
+		/**
+		 * Adds an item into the collection.
+		 *
+		 * Note that this is an array-like collection, so the same item can be present more than once. This behavior is
+		 * for performance purposes only and is not guaranteed to be kept in the same way in the future.
+		 *
+		 * @param {Model} model The item to be added.
+		 */
+		add: function( model ) {
+			model.addParentEmitter( this );
```

Comment:
> Then, the critical part... listenTo what? We would have to introduce a new all event just for that, which would have an even bigger impact on the API.

True, that's why I'm a bit scared of bubbling all kinds of events from a child. It may eventually happen that a child will  try to fire an event with the same name the parent (collection) uses and give us a huge headache debugging it :) That's why I'd still go with Backbone's approach and just bubble child's `change` event.